hazelcast/src/test/java/com/hazelcast/jet/impl/deployment/NonSmartClientTest.java

```diff
@@ -198,12 +199,17 @@ public void when_lightJobSubmittedToNonMaster_then_accessibleFromAllMembers() {
         assertEquals(job1.getSubmissionTime(), job2.getSubmissionTime());
         assertFalse(job1.getFuture().isDone());
         assertFalse(job2.getFuture().isDone());
-        // cancel requested through client connected to the master, but job is coordinated by the other member
+        // Cancel requested through client connected to the master, but job is coordinated by the other member.
+        // The jobX.getFuture() invokes JoinSubmittedJobOperation on a member asynchronously. The jobX.cancel()
+        // invokes TerminateJobOperation. We have no control over which of this operation's doRun() method invokes
+        // first. If first is JoinSubmittedJobOperation then the join() throws CancellationException. If first is
+        // TerminateJobOperation (that results later in removing of light master context) then the join() throws
+        // CompletionException.
         job1.cancel();
         try {
             job1.join();
             fail("join didn't fail");
-        } catch (CancellationException ignored) { }
+        } catch (CancellationException | CompletionException ignored) { }
```

Comment:
Every exception other than `CancellationException`  is wrapped in a `CompletionException`. We should check for the `JobNotFoundException`, which is the cause of the `CompletionException`.

Reply:
Good point.